device/cuda/src/ambiguity_resolution/greedy_ambiguity_resolution_algorithm.cu

```diff
@@ -542,19 +543,19 @@ greedy_ambiguity_resolution_algorithm::operator()(
 
         // TODO: Make n_it adaptive based on the average track length, bound
         // value in count_removable_tracks, etc.
-        const unsigned int n_it = 100;
-        for (unsigned int iter = 0; iter < n_it; iter++) {
-            cudaGraphLaunch(graphExec, stream);
-        }
+        cudaGraphLaunch(graphExec, stream);
 
         cudaMemcpyAsync(&terminate, terminate_device.get(), sizeof(int),
                         cudaMemcpyDeviceToHost, stream);
         cudaMemcpyAsync(&n_accepted, n_accepted_device.get(),
                         sizeof(unsigned int), cudaMemcpyDeviceToHost, stream);
+        m_stream.get().synchronize();
```

Comment:
I also don't use a synchronize function as its overhead was pretty big. Have you encountered any issue regarding it?

Reply:
It's good that you pointed this out Beomki.

The deeper issue here is that using `cudaMemcpyAsync` suggests that something would be happening asynchronously here. But since you're copying into variables that are on the stack, these copies will very much happen synchronously. (It's a long story with how SYCL will happily set up asynchronous copies into/from temporary memory areas, but CUDA is only willing to do so with "pinned" host memory. I believe CUDA is the one behaving in the expected way here.)

So indeed, the explicit synchronization should not be doing anything here. But at the same time, it would also probably be worth making the memory copies into synchronous ones as well, to avoid this type of confusion in the future.